demos/fleet_provisioning/fleet_provisioning_with_csr/fleet_provisioning_csr_demo.c

```diff
@@ -174,7 +172,8 @@ static size_t thingNameLength;
 static char payloadBuffer[ NETWORK_BUFFER_SIZE ];
 
 /**
- * @brief Length of the payload stored in #payloadBuffer.
+ * @brief Length of the payload stored in #payloadBuffer. This is set by the
+ * MQTT publish callback when it copies a received payload into #payloadBuffer.
```

Comment:
(Doc Nit) Same comment can be added to the `payloadBuffer` global about it being updated the with payload data by the publish callback